pkg/agent/openflow/regmarks.go

```diff
@@ -0,0 +1,109 @@
+	MarkCustomReasonLogging = binding.NewRegMarkWithField(CustomReasonField, CustomReasonLogging)
+	MarkCustomReasonReject  = binding.NewRegMarkWithField(CustomReasonField, CustomReasonReject)
+	MarkCustomReasonDeny    = binding.NewRegMarkWithField(CustomReasonField, CustomReasonDeny)
+
+	// reg1(NXM_NX_REG1)
+	// Field to cache the ofPort of the OVS interface where to output packet.
+	TargetOFPortField = binding.NewRegField(1, 0, 31)
+
+	// reg2(NXM_NX_REG2)
+	// Field to help swap values in two different flow fields in the OpenFlow actions. This field is only used in func
+	// `arpResponderStaticFlow`.
+	SwapField = binding.NewRegField(2, 0, 31)
+
+	// reg3(NXM_NX_REG3)
+	// Field to store the selected Service Endpoint IP
+	EndpointIPField = binding.NewRegField(3, 0, 31)
+	// Field to store the conjunction ID which is for "deny" rule in CNP. It shares the same register with EndpointIPField,
+	// since the service selection will finish when a packet hitting NetworkPolicy related rules.
+	CNPDenyConjIDField = binding.NewRegField(3, 0, 31)
+
+	// reg4(NXM_NX_REG4)
+	// reg4[0..15]: Field to store the selected Service Endpoint port.
+	EndpointPortField = binding.NewRegField(4, 0, 15)
+	// reg4[16..18]: Field to store the state of a packet accessing a Service. Marks in this field include,
+	//	- 0b001: packet need to do service selection.
+	//	- 0b010: packet has done service selection.
+	//	- 0b011: packet has done service selection and the selection result needs to be cached.
+	ServiceEPStateField = binding.NewRegField(4, 16, 18)
+	EpToSelectMark      = binding.NewRegMarkWithField(ServiceEPStateField, 0b001)
+	EpSelectedMark      = binding.NewRegMarkWithField(ServiceEPStateField, 0b010)
+	EpToLearnMark       = binding.NewRegMarkWithField(ServiceEPStateField, 0b011)
+	// reg4[0..18]: Field to cache the union value of Endpoint port and Endpoint status. It is used as a single match
+	// when needed.
+	EpUnionField = binding.NewRegField(4, 0, 18)
+
+	// reg5(NXM_NX_REG5)
+	// Field to cache the Egress conjunction ID hit by TraceFlow packet.
+	TFEgressConjIDField = binding.NewRegField(5, 0, 31)
+
+	// reg(N6XM_NX_REG6)
+	// Field to cache the Ingress conjunction ID hit by TraceFlow packet.
+	TFIngressConjIDField = binding.NewRegField(6, 0, 31)
+)
+
+// Fields used xxreg.
+var (
+	// xxreg3(NXM_NX_XXREG3)
+	// xxreg3: Field to cache Endpoint IPv6 address. It occupies reg12-reg15 in the meanwhile.
+	EndpointIP6Field = binding.NewXXRegField(3, 0, 127)
+)
+
+// Marks used in conntrack
+var (
+	// Mark to indicate the connection is initiated through the host gateway interface
+	// (i.e. for which the first packet of the connection was received through the gateway).
+	gatewayCTMark = binding.NewCtMark(0x20, 0, 31)
+	// Mark to indicate DNAT is performed on the connection for Service.
+	ServiceCTMark = binding.NewCtMark(0x21, 0, 31)
+)
+
+func createOneBitRegMark(regID int, bit uint32) *binding.RegMark {
```

Comment:
Would you name it newOneBitRegMark to be consistent with others, and even move it to binding?